### ISSUE TITLE
[wip]Google Code-In 2019-20 - Pending Issues

### ISSUE THREAD

**Issue by CodeSarthak** [publiclab/plots2#7014]
This is a staging area. Each task has (or will soon have) a link to a full issue including relevant lines of code, screenshots, and other helpful information as we prepare them for solving. Some are harder than others, but those tagged with `first-timers-only` are great to begin with. (more [documentation on tasks](https://developers.google.com/open-source/gci/help/org-sample-tasks))

Add new items: Look here for new items to adapt for this list: [planning issues](https://github.com/publiclab/plots2/labels/planning), [break-me-up issues](https://github.com/publiclab/plots2/labels/break-me-up), [fto-candidates](https://github.com/publiclab/plots2/labels/fto-candidate). Fit tasks into the categories below.

Format items to be descriptive, `[bracket otherwise]`: As we create this list, it's OK to be brief, but if a task lacks description, tags, screenshots, and code links, `[Put it in brackets to indicate that it needs to be fleshed out]`

Break out items here into their own issues: Use the `gci-candidate` tag instead of the `gci` tag to manage the supply. We don't want to have more open issues than we can support at one time, and `gci` is what we'll direct students towards. `gci-candidate` is for "held back" issues we're saving up.

Weighting: some issues don't result in many lines of code, but are still big projects. Use tags `small`, `medium`, `big` when you'd like to help mark them ([read more](https://github.com/publiclab/plots2/issues/3284#issuecomment-422071860))

-   [ ]  display author profile bio below content in moderator email notifications: <https://github.com/publiclab/plots2/blob/master/app/views/admin_mailer/notify_node_moderators.html.erb>
-   [ ]  complete 'sortable headers' implementation following pattern in [#2352](https://github.com/publiclab/plots2/issues/2352)
-   [ ]  additional work on the Checklists system: [#2452](https://github.com/publiclab/plots2/issues/2452)
-   [ ]  calculate "answered" rate for questions posted in a certain time period (described in [#2120](https://github.com/publiclab/plots2/issues/2120))

These can be moved up once they have code links or issue links:

-   [ ]  `needs-code-links` Preserve redirect to original destination when accessing `/post` while not logged in (follow these steps: 1. publiclab.org/post You will be redirected to login page 2. Login 3. Redirected to dashboard -- Expected Behavior Redirected to /post.)
-   [ ]  display of an SVG at the URL `/profile/status/_ID_` for a `user` with `id` of `_ID_`, showing the `user.status` as one of these two SVGs: <https://gist.github.com/jywarren/ea0523a09f6f8660486b14eeea15de95>
-   [ ]  make username/email form appear side by side with Oauth user registration at <https://publiclab.org/signup>

-   [ ]  `needs-code-links` Add generic video icon here, instead of being specific to YouTube. We should give equal priority to all the hosts. (<https://github.com/publiclab/PublicLab.Editor/>)\
    [![image](https://user-images.githubusercontent.com/20972099/45542852-a9b5b900-b830-11e8-9101-b5085c6565d3.png)](https://user-images.githubusercontent.com/20972099/45542852-a9b5b900-b830-11e8-9101-b5085c6565d3.png)
-   [ ]  On clicking on '?' page should be opened in new tab. Currently, it is opening on the same page, using `target="_blank"` on the `<a>` tag\
    [![image](https://user-images.githubusercontent.com/20972099/45542912-d79afd80-b830-11e8-95d9-4bbfb533c1cd.png)](https://user-images.githubusercontent.com/20972099/45542912-d79afd80-b830-11e8-95d9-4bbfb533c1cd.png) -- `GCI dashboard`
-   [ ]  `needs-code-links` Similar behaviour (needs to open on a new page) reported for `Open Source Your Work` link at <https://publiclab.org/signup> and /post
-   [ ]  `needs-code-links` Hover text should redirect to page listing all the wiki articles and research notes with text `water quality`\
    [![image](https://user-images.githubusercontent.com/20972099/45543039-35c7e080-b831-11e8-8d25-bc060801a490.png)](https://user-images.githubusercontent.com/20972099/45543039-35c7e080-b831-11e8-8d25-bc060801a490.png)
-   [ ]  `needs-code-links`Hover text should redirect to page listing all the wiki articles and research notes with text `balloon-mapping`\
    [![image](https://user-images.githubusercontent.com/20972099/45543048-382a3a80-b831-11e8-8f1b-e10dc81ee4de.png)](https://user-images.githubusercontent.com/20972099/45543048-382a3a80-b831-11e8-8f1b-e10dc81ee4de.png)
-   [ ]  `needs-code-links` Clearly mention that on RHS we have Research notes. `Liked by ____` appears ambiguous to me a lot\
    [![image](https://user-images.githubusercontent.com/20972099/45543333-ec2bc580-b831-11e8-82f1-4e3e216d1914.png)](https://user-images.githubusercontent.com/20972099/45543333-ec2bc580-b831-11e8-82f1-4e3e216d1914.png)
-   [ ]  `needs-code-links` sort wiki page likes according to date/time\
    [![image](https://user-images.githubusercontent.com/20972099/45543424-06fe3a00-b832-11e8-9ec9-32e564306ee6.png)](https://user-images.githubusercontent.com/20972099/45543424-06fe3a00-b832-11e8-9ec9-32e564306ee6.png)
-   [ ]  `needs-code-links` If a person has no barnstars and they click on the nav bar. There must be a message to show them. Just to enhance the readability\
    [![image](https://user-images.githubusercontent.com/20972099/45543539-5d6b7880-b832-11e8-8b2b-33cdf3c28ebf.png)](https://user-images.githubusercontent.com/20972099/45543539-5d6b7880-b832-11e8-8b2b-33cdf3c28ebf.png)
-   [ ]  On removal of any of these tags, all three must be removed [![image](https://user-images.githubusercontent.com/20972099/45543633-9c99c980-b832-11e8-9e50-0c52fa207118.png)](https://user-images.githubusercontent.com/20972099/45543633-9c99c980-b832-11e8-9e50-0c52fa207118.png)
-   [ ]  Same button three times to post work on same web page [![image](https://user-images.githubusercontent.com/20972099/45563088-e4881300-b869-11e8-87e2-c9108fb6f5c9.png)](https://user-images.githubusercontent.com/20972099/45563088-e4881300-b869-11e8-87e2-c9108fb6f5c9.png)
-   [ ]  Same thread is placed repeatedly on the dashboard page due to multiple comments.\
    [![image](https://user-images.githubusercontent.com/20972099/45541563-1464f580-b82d-11e8-98ca-c5b7759389ad.png)](https://user-images.githubusercontent.com/20972099/45541563-1464f580-b82d-11e8-98ca-c5b7759389ad.png)\
    I will suggest that one post should be shown only once.
-   [ ]  Standardize display of `n` Wiki and `m` Research Notes [#2193](https://github.com/publiclab/plots2/issues/2193)
-   [ ]  Test for the the above task
-   [ ]  Tag the author on the home page where their articles are written in grid view/List view (needs own issue)
-   [ ]  logging users to wiki page when a user logs in via header on wiki page (needs own issue)
-   [ ]  Clicking on like button if not logged in should redirect to login page. (needs own issue)
-   [ ]  Pagination broken on home page\
    [![image](https://user-images.githubusercontent.com/20972099/45540558-62c4c500-b82a-11e8-92a9-53dfe375afbc.png)](https://user-images.githubusercontent.com/20972099/45540558-62c4c500-b82a-11e8-92a9-53dfe375afbc.png)\
    see that url shows page 2 but the pagination is of page 1. (needs own issue)
-   [ ]  `needs-code-links` make tags in `Suggested tags` on publiclab.org/post appear in tag input box when clicked - <https://github.com/publiclab/PublicLab.Editor/>

Documentation/Training: creating/editing documents and helping others learn more
--------------------------------------------------------------------------------

-   [ ]  write a document explaining how to start contributing at Public Lab for RGSoc (needs own issue)

-   [ ]  the table of contents on the github.com/publiclab/plots2 is inconsistent with the content present after it (needs own issue)

Outreach/Research: community management, marketing, or studying problems and recommending solutions
---------------------------------------------------------------------------------------------------

-   [ ]  embed our welcome page in full at <https://publiclab.org/code> (needs own issue)
-   [ ]  read through welcome page <https://publiclab.github.io/community-toolbox/> and highlight gaps or ambiguities (needs own issue in <https://github.com/publiclab/community-toolbox/issues>)

Quality Assurance: testing and ensuring code is of high quality
---------------------------------------------------------------

-   [ ]  [#2151](https://github.com/publiclab/plots2/issues/2151) - Fix "Show 0 more" on inline people lists

-   [ ]  Read through Code Climate "smells" and open issues related to solving them with code links included and recommendations: <https://codeclimate.com/github/publiclab/plots2/issues?severity%5B%5D=critical&severity%5B%5D=major&status%5B%5D=&status%5B%5D=open&status%5B%5D=confirmed&engine_name%5B%5D=structure&engine_name%5B%5D=duplication> (described in [#3706](https://github.com/publiclab/plots2/issues/3706)) -- `GCI dashboard`.

Design: graphic design or user interface design
-----------------------------------------------

-   [ ]  BIG TASK: Wiki page should be rendered on same page. On clicking on wiki pages nav bar on /notes/popular a new page opens up. But wiki pages should be rendered under the wiki nav bar. [#3456](https://github.com/publiclab/plots2/issues/3456)
-   [ ]  FTO: make the rsvp links in the events on the home page non clickable.
-   [ ]  [![image](https://user-images.githubusercontent.com/20972099/45532373-6bf56800-b811-11e8-9be0-ba0aa16d1905.png)](https://user-images.githubusercontent.com/20972099/45532373-6bf56800-b811-11e8-9be0-ba0aa16d1905.png)\
    ''Read More'' should be center aligned
-   [ ]  "Read more" and the next grid item should have hr in between\
    [![image](https://user-images.githubusercontent.com/20972099/45532417-92b39e80-b811-11e8-9f86-b69bc9b417c6.png)](https://user-images.githubusercontent.com/20972099/45532417-92b39e80-b811-11e8-9f86-b69bc9b417c6.png)

-   [ ]  [multiple-user]Copy welcomebot and first-timers-only templates into other projects - [#3681](https://github.com/publiclab/plots2/issues/3681) -- `GCI dashboard`

**Comment by SidharthBansal** [comment]
Can you please check the list and write the following comments against each of the issue?
1) [Uploaded]: if it is present on GCI website
2) [Needs new issue]: If issue is present but there are really long conversations. Hence we need to reformat it to a new issue.
3) [Create new issue]: If no issue is present yet
4) [Not uploaded]: Not present on GCI dashboard
etc.

Once done, then we can move forward with this
Thanks a lot!

**Comment by SidharthBansal** [comment]
Just a suggestion, please check the bugs written above on the actual website. I hope many things are fixed
Thanks

**Comment by SidharthBansal** [comment]
logging users to wiki page when a user logs in via header on wiki page (needs own issue)

  Clicking on like button if not logged in should redirect to login page. (needs own issue)

  write a document explaining how to start contributing at Public Lab for RGSoc (needs own issue)

needs-code-links Preserve redirect to original destination when accessing /post while not logged in (follow these steps: 1. publiclab.org/post You will be redirected to login page 2. Login 3. Redirected to dashboard -- Expected Behavior Redirected to /post.)


 make username/email form appear side by side with Oauth user registration at https://publiclab.org/signup


 needs-code-links Similar behavior (needs to open on a new page) reported for Open Source Your Work link at https://publiclab.org/signup and /post

I think all of these issues are solved. Please check prod and remove things that are already solved.
Also, I like to tell you that the dashboard, profile page, and login-signup have changed completely from the last year. So, many issues are of no use this year. Kindly remove them

**Comment by CodeSarthak** [comment]
I am sorry, I will check the website to get clarification on the pending issues. Further, I will write the comments against each task/issue. 

Thank you.

**Comment by Uzay-G** [comment]
Refactored in #7408
